src/is_parser.c

```diff
@@ -82,6 +75,10 @@ is_parse(ISParser *parser)
             SKIP_SPACES(parser->ptr);
             SKIP_ESCAPED(parser->ptr);
             GET_NUM(parser->ptr, key, parser->begin);
+
+            if (key < PG_INT32_MIN || key > PG_INT32_MAX)
```

Comment:
`GET_NUM` is used below for another `key` variable in `is_parse_arr()` function which is still `int32`. Probably it is better to change its type as well and check for overflow or just add new macros for `int32` types.

Reply:
True, i didn't notice the `is_parse_arr()` func. I'll fix it.

Reply:
I fixed this. Also I regenerated test scripts (using dumbo) which caused a lot of noise (changed line numbers).